### ISSUE TITLE
vec_init_then_push fires for loop immediately after vector creation

### ISSUE THREAD

**Issue by sunshowers** [rust-lang/rust-clippy#6956]
(On Rust 1.51.0-beta.8).

I tried this code:

```rust
fn main() {
    let mut x = vec![];
    loop {
        x.push("test");
    }
}
```

I expected to see this happen: no error

Instead, this happened:

```
warning: calls to `push` immediately after creation
 --> src/main.rs:2:5
  |
2 | /     let mut x = vec![];
3 | |     loop {
4 | |         x.push("test");
  | |_______________________^ help: consider using the `vec![]` macro: `let mut x = vec![..];`
  |
  = note: `#[warn(clippy::vec_init_then_push)]` on by default
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#vec_init_then_push

warning: 1 warning emitted
```

### Meta

- `cargo clippy -V`: clippy 0.1.51 (73f48e5 2021-03-18)
- `rustc -Vv`:

```
rustc 1.51.0-beta.8 (73f48e5f6 2021-03-18)
binary: rustc
commit-hash: 73f48e5f6f6217c4bf93026bc0072637966d6eab
commit-date: 2021-03-18
host: x86_64-unknown-linux-gnu
release: 1.51.0-beta.8
LLVM version: 11.0.1
```

**Comment by Manishearth** [comment]
Yeah we should probably ignore loops

**Comment by Manishearth** [comment]
And most kinds of conditionals

**Comment by Y-Nak** [comment]
This is fixed in #6697
